File: java/common/src/main/java/com/ais/middleware/common/events/prs/PLUWAppraisalCreatedEvent.java
package com.ais.middleware.common.events.prs;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;

/**
 * Published by PLUWGateway stub after successfully creating the appraisal/inspection record.
 * Topic: prs.events.pluw-appraisal-created
 *
 * ⚠️ DEMO GAP: Real PLUW response fields not confirmed — pluwReferenceId is fabricated.
 */
public record PLUWAppraisalCreatedEvent(
        String correlationId,
        // ⚠️ DEMO GAP: Actual PLUW reference ID format unknown
        String pluwReferenceId,
        boolean success,
        OffsetDateTime createdAt
) {
    @JsonCreator
    public PLUWAppraisalCreatedEvent(
            @JsonProperty("correlationId") String correlationId,
            @JsonProperty("pluwReferenceId") String pluwReferenceId,
            @JsonProperty("success") boolean success,
            @JsonProperty("createdAt") OffsetDateTime createdAt
    ) {
        this.correlationId = correlationId;
        this.pluwReferenceId = pluwReferenceId;
        this.success = success;
        this.createdAt = createdAt;
    }
}

File: dotnet/Middleware.Contracts/Events/PLUWAppraisalCreatedEvent.cs
namespace Middleware.Contracts.Events;

// UC4 · Appraisal Documents — PLUW appraisal/inspection created (parallel branch result)
public sealed class PLUWAppraisalCreatedEvent
{
    public string AppraisalId { get; set; } = string.Empty;

    // ⚠️ DEMO GAP: PLUW work item ID format unknown — REPLACE_ME_PLUW_WORK_ITEM_FORMAT
    public string PLUWWorkItemId { get; set; } = string.Empty;

    // ⚠️ DEMO GAP: @Work inspection ID format unknown — REPLACE_ME_ATWORK_INSPECTION_ID
    public string AtWorkInspectionId { get; set; } = string.Empty;

    public string CorrelationId { get; set; } = string.Empty;
    public DateTimeOffset CreatedAt { get; set; }
}
